<u><b>PROBLEM STATEMENT</u></b>

Your company is expanding into new industries to diversify its portfolio. Specifically, 
1. <i>Interested in purchasing and operating airplanes for commercial and private enterprises</i> 
2. <i>Do not know anything about the potential risks of aircraft.</i>
3. <i>Determining which aircraft are the lowest risk for the company to start this new business endeavor.</i>
4. <i>Translate your findings into actionable insights to help decide which aircraft to purchase.</i>

In [1]:
# importing needed modules
import pandas as pd
import csv

In [2]:
# reading the csv file
df = pd.read_csv('Aviation_Data.csv')
df

c:\Users\SHIRU\anaconda3\envs\learn-env\lib\site-packages\IPython\core\interactiveshell.py:3145: DtypeWarning: Columns (6,7,28) have mixed types.Specify dtype option on import or set low_memory=False.
  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,


Event.Id Investigation.Type Accident.Number  Event.Date  \
0      20001218X45444           Accident      SEA87LA080  1948-10-24   
1      20001218X45447           Accident      LAX94LA336  1962-07-19   
2      20061025X01555           Accident      NYC07LA005  1974-08-30   
3      20001218X45448           Accident      LAX96LA321  1977-06-19   
4      20041105X01764           Accident      CHI79FA064  1979-08-02   
...               ...                ...             ...         ...   
90343  20221227106491           Accident      ERA23LA093  2022-12-26   
90344  20221227106494           Accident      ERA23LA095  2022-12-26   
90345  20221227106497           Accident      WPR23LA075  2022-12-26   
90346  20221227106498           Accident      WPR23LA076  2022-12-26   
90347  20221230106513           Accident      ERA23LA097  2022-12-29   

              Location        Country Latitude Longitude Airport.Code  \
0      MOOSE CREEK, ID  United States      NaN       NaN          NaN   
1       BRIDGEPORT, CA  United States      NaN       NaN          NaN   
2        Saltville, VA  United States  36.9222  -81.8781          NaN   
3           EUREKA, CA  United States      NaN       NaN          NaN   
4           Canton, OH  United States      NaN       NaN          NaN   
...                ...            ...      ...       ...          ...   
90343    Annapolis, MD  United States      NaN       NaN          NaN   
90344      Hampton, NH  United States      NaN       NaN          NaN   
90345       Payson, AZ  United States  341525N  1112021W          PAN   
90346       Morgan, UT  United States      NaN       NaN          NaN   
90347       Athens, GA  United States      NaN       NaN          NaN   

      Airport.Name  ... Purpose.of.flight         Air.carrier  \
0              NaN  ...          Personal                 NaN   
1              NaN  ...          Personal                 NaN   
2              NaN  ...          Personal                 NaN   
3              NaN  ...          Personal                 NaN   
4              NaN  ...          Personal                 NaN   
...            ...  ...               ...                 ...   
90343          NaN  ...          Personal                 NaN   
90344          NaN  ...               NaN                 NaN   
90345       PAYSON  ...          Personal                 NaN   
90346          NaN  ...          Personal  MC CESSNA 210N LLC   
90347          NaN  ...          Personal                 NaN   

      Total.Fatal.Injuries Total.Serious.Injuries Total.Minor.Injuries  \
0                      2.0                    0.0                  0.0   
1                      4.0                    0.0                  0.0   
2                      3.0                    NaN                  NaN   
3                      2.0                    0.0                  0.0   
4                      1.0                    2.0                  NaN   
...                    ...                    ...                  ...   
90343                  0.0                    1.0                  0.0   
90344                  0.0                    0.0                  0.0   
90345                  0.0                    0.0                  0.0   
90346                  0.0                    0.0                  0.0   
90347                  0.0                    1.0                  0.0   

      Total.Uninjured Weather.Condition  Broad.phase.of.flight  \
0                 0.0               UNK                 Cruise   
1                 0.0               UNK                Unknown   
2                 NaN               IMC                 Cruise   
3                 0.0               IMC                 Cruise   
4                 0.0               VMC               Approach   
...               ...               ...                    ...   
90343             0.0               NaN                    NaN   
90344             0.0               NaN                    NaN   
90345  

In [3]:
# data info
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 90348 entries, 0 to 90347
Data columns (total 31 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Event.Id                88889 non-null  object 
 1   Investigation.Type      90348 non-null  object 
 2   Accident.Number         88889 non-null  object 
 3   Event.Date              88889 non-null  object 
 4   Location                88837 non-null  object 
 5   Country                 88663 non-null  object 
 6   Latitude                34382 non-null  object 
 7   Longitude               34373 non-null  object 
 8   Airport.Code            50249 non-null  object 
 9   Airport.Name            52790 non-null  object 
 10  Injury.Severity         87889 non-null  object 
 11  Aircraft.damage         85695 non-null  object 
 12  Aircraft.Category       32287 non-null  object 
 13  Registration.Number     87572 non-null  object 
 14  Make                    88826 non-null

In [ ]:
# checking for duplicates
df.duplicated().sum()

0

In [4]:
# checking for null values
df.isnull().sum()

Event.Id                   1459
Investigation.Type            0
Accident.Number            1459
Event.Date                 1459
Location                   1511
Country                    1685
Latitude                  55966
Longitude                 55975
Airport.Code              40099
Airport.Name              37558
Injury.Severity            2459
Aircraft.damage            4653
Aircraft.Category         58061
Registration.Number        2776
Make                       1522
Model                      1551
Amateur.Built              1561
Number.of.Engines          7543
Engine.Type                8536
FAR.Description           58325
Schedule                  77766
Purpose.of.flight          7651
Air.carrier               73700
Total.Fatal.Injuries      12860
Total.Serious.Injuries    13969
Total.Minor.Injuries      13392
Total.Uninjured            7371
Weather.Condition          5951
Broad.phase.of.flight     28624
Report.Status              7840
Publication.Date          16689
dtype: i

In [5]:
# removing the null values in the event.id column cause it is the primary key
df.dropna(subset=['Event.Id'], inplace=True)
df

Event.Id Investigation.Type Accident.Number  Event.Date  \
0      20001218X45444           Accident      SEA87LA080  1948-10-24   
1      20001218X45447           Accident      LAX94LA336  1962-07-19   
2      20061025X01555           Accident      NYC07LA005  1974-08-30   
3      20001218X45448           Accident      LAX96LA321  1977-06-19   
4      20041105X01764           Accident      CHI79FA064  1979-08-02   
...               ...                ...             ...         ...   
90343  20221227106491           Accident      ERA23LA093  2022-12-26   
90344  20221227106494           Accident      ERA23LA095  2022-12-26   
90345  20221227106497           Accident      WPR23LA075  2022-12-26   
90346  20221227106498           Accident      WPR23LA076  2022-12-26   
90347  20221230106513           Accident      ERA23LA097  2022-12-29   

              Location        Country Latitude Longitude Airport.Code  \
0      MOOSE CREEK, ID  United States      NaN       NaN          NaN   
1       BRIDGEPORT, CA  United States      NaN       NaN          NaN   
2        Saltville, VA  United States  36.9222  -81.8781          NaN   
3           EUREKA, CA  United States      NaN       NaN          NaN   
4           Canton, OH  United States      NaN       NaN          NaN   
...                ...            ...      ...       ...          ...   
90343    Annapolis, MD  United States      NaN       NaN          NaN   
90344      Hampton, NH  United States      NaN       NaN          NaN   
90345       Payson, AZ  United States  341525N  1112021W          PAN   
90346       Morgan, UT  United States      NaN       NaN          NaN   
90347       Athens, GA  United States      NaN       NaN          NaN   

      Airport.Name  ... Purpose.of.flight         Air.carrier  \
0              NaN  ...          Personal                 NaN   
1              NaN  ...          Personal                 NaN   
2              NaN  ...          Personal                 NaN   
3              NaN  ...          Personal                 NaN   
4              NaN  ...          Personal                 NaN   
...            ...  ...               ...                 ...   
90343          NaN  ...          Personal                 NaN   
90344          NaN  ...               NaN                 NaN   
90345       PAYSON  ...          Personal                 NaN   
90346          NaN  ...          Personal  MC CESSNA 210N LLC   
90347          NaN  ...          Personal                 NaN   

      Total.Fatal.Injuries Total.Serious.Injuries Total.Minor.Injuries  \
0                      2.0                    0.0                  0.0   
1                      4.0                    0.0                  0.0   
2                      3.0                    NaN                  NaN   
3                      2.0                    0.0                  0.0   
4                      1.0                    2.0                  NaN   
...                    ...                    ...                  ...   
90343                  0.0                    1.0                  0.0   
90344                  0.0                    0.0                  0.0   
90345                  0.0                    0.0                  0.0   
90346                  0.0                    0.0                  0.0   
90347                  0.0                    1.0                  0.0   

      Total.Uninjured Weather.Condition  Broad.phase.of.flight  \
0                 0.0               UNK                 Cruise   
1                 0.0               UNK                Unknown   
2                 NaN               IMC                 Cruise   
3                 0.0               IMC                 Cruise   
4                 0.0               VMC               Approach   
...               ...               ...                    ...   
90343             0.0               NaN                    NaN   
90344             0.0               NaN                    NaN   
90345  

In [6]:
df.isna().sum()

Event.Id                      0
Investigation.Type            0
Accident.Number               0
Event.Date                    0
Location                     52
Country                     226
Latitude                  54507
Longitude                 54516
Airport.Code              38640
Airport.Name              36099
Injury.Severity            1000
Aircraft.damage            3194
Aircraft.Category         56602
Registration.Number        1317
Make                         63
Model                        92
Amateur.Built               102
Number.of.Engines          6084
Engine.Type                7077
FAR.Description           56866
Schedule                  76307
Purpose.of.flight          6192
Air.carrier               72241
Total.Fatal.Injuries      11401
Total.Serious.Injuries    12510
Total.Minor.Injuries      11933
Total.Uninjured            5912
Weather.Condition          4492
Broad.phase.of.flight     27165
Report.Status              6381
Publication.Date          15230
dtype: i

For a client to purchase an aircraft the type of injury that occurs using a specific aircraft matters, so we are going to see the types of injuries that occur and remove the null values on it

In [25]:
# checking for the severity values
severity = df.loc[:, 'Injury.Severity']
severity

0         Fatal(2)
1         Fatal(4)
2         Fatal(3)
3         Fatal(2)
4         Fatal(1)
           ...    
90343        Minor
90344      Unknown
90345    Non-Fatal
90346      Unknown
90347        Minor
Name: Injury.Severity, Length: 88889, dtype: object

In [ ]:
# replacing the non values with unknown
df['Injury.Severity'] = df['Injury.Severity'].fillna('Unknown')
df

Event.Id Investigation.Type Accident.Number  Event.Date  \
0      20001218X45444           Accident      SEA87LA080  1948-10-24   
1      20001218X45447           Accident      LAX94LA336  1962-07-19   
2      20061025X01555           Accident      NYC07LA005  1974-08-30   
3      20001218X45448           Accident      LAX96LA321  1977-06-19   
4      20041105X01764           Accident      CHI79FA064  1979-08-02   
...               ...                ...             ...         ...   
90343  20221227106491           Accident      ERA23LA093  2022-12-26   
90344  20221227106494           Accident      ERA23LA095  2022-12-26   
90345  20221227106497           Accident      WPR23LA075  2022-12-26   
90346  20221227106498           Accident      WPR23LA076  2022-12-26   
90347  20221230106513           Accident      ERA23LA097  2022-12-29   

              Location        Country Latitude Longitude Airport.Code  \
0      MOOSE CREEK, ID  United States      NaN       NaN          NaN   
1       BRIDGEPORT, CA  United States      NaN       NaN          NaN   
2        Saltville, VA  United States  36.9222  -81.8781          NaN   
3           EUREKA, CA  United States      NaN       NaN          NaN   
4           Canton, OH  United States      NaN       NaN          NaN   
...                ...            ...      ...       ...          ...   
90343    Annapolis, MD  United States      NaN       NaN          NaN   
90344      Hampton, NH  United States      NaN       NaN          NaN   
90345       Payson, AZ  United States  341525N  1112021W          PAN   
90346       Morgan, UT  United States      NaN       NaN          NaN   
90347       Athens, GA  United States      NaN       NaN          NaN   

      Airport.Name  ... Purpose.of.flight         Air.carrier  \
0              NaN  ...          Personal                 NaN   
1              NaN  ...          Personal                 NaN   
2              NaN  ...          Personal                 NaN   
3              NaN  ...          Personal                 NaN   
4              NaN  ...          Personal                 NaN   
...            ...  ...               ...                 ...   
90343          NaN  ...          Personal                 NaN   
90344          NaN  ...               NaN                 NaN   
90345       PAYSON  ...          Personal                 NaN   
90346          NaN  ...          Personal  MC CESSNA 210N LLC   
90347          NaN  ...          Personal                 NaN   

      Total.Fatal.Injuries Total.Serious.Injuries Total.Minor.Injuries  \
0                      2.0                    0.0                  0.0   
1                      4.0                    0.0                  0.0   
2                      3.0                    NaN                  NaN   
3                      2.0                    0.0                  0.0   
4                      1.0                    2.0                  NaN   
...                    ...                    ...                  ...   
90343                  0.0                    1.0                  0.0   
90344                  0.0                    0.0                  0.0   
90345                  0.0                    0.0                  0.0   
90346                  0.0                    0.0                  0.0   
90347                  0.0                    1.0                  0.0   

      Total.Uninjured Weather.Condition  Broad.phase.of.flight  \
0                 0.0               UNK                 Cruise   
1                 0.0               UNK                Unknown   
2                 NaN               IMC                 Cruise   
3                 0.0               IMC                 Cruise   
4                 0.0               VMC               Approach   
...               ...               ...                    ...   
90343             0.0               NaN                    NaN   
90344             0.0               NaN                    NaN   
90345  

In [11]:
# rechecking for the severity values to see if the null values have been replaced
severity = df.loc[:, 'Injury.Severity']
severity

0         Fatal(2)
1         Fatal(4)
2         Fatal(3)
3         Fatal(2)
4         Fatal(1)
           ...    
90343        Minor
90344      Unknown
90345    Non-Fatal
90346      Unknown
90347        Minor
Name: Injury.Severity, Length: 88889, dtype: object

Now that we have handled the Injury.Severity column, next is the different type of injuries(Fatal, serious, Minor and uninjures) and since they are quantitative variables, we replace the null values with 0 instead on dropping them

In [12]:
# replacing the null values on the Total.fatal.injuries column, total.minor.injuries and total.serious.injuries Total.Uninjured
df['Total.Fatal.Injuries'] = df['Total.Fatal.Injuries'].fillna(0)
df['Total.Minor.Injuries'] = df['Total.Minor.Injuries'].fillna(0)
df['Total.Serious.Injuries'] = df['Total.Serious.Injuries'].fillna(0)
df['Total.Uninjured'] = df['Total.Uninjured'].fillna(0)
df


The history saving thread hit an unexpected error (OperationalError('database or disk is full')).History will not be written to the database.


Event.Id Investigation.Type Accident.Number  Event.Date  \
0      20001218X45444           Accident      SEA87LA080  1948-10-24   
1      20001218X45447           Accident      LAX94LA336  1962-07-19   
2      20061025X01555           Accident      NYC07LA005  1974-08-30   
3      20001218X45448           Accident      LAX96LA321  1977-06-19   
4      20041105X01764           Accident      CHI79FA064  1979-08-02   
...               ...                ...             ...         ...   
90343  20221227106491           Accident      ERA23LA093  2022-12-26   
90344  20221227106494           Accident      ERA23LA095  2022-12-26   
90345  20221227106497           Accident      WPR23LA075  2022-12-26   
90346  20221227106498           Accident      WPR23LA076  2022-12-26   
90347  20221230106513           Accident      ERA23LA097  2022-12-29   

              Location        Country Latitude Longitude Airport.Code  \
0      MOOSE CREEK, ID  United States      NaN       NaN          NaN   
1       BRIDGEPORT, CA  United States      NaN       NaN          NaN   
2        Saltville, VA  United States  36.9222  -81.8781          NaN   
3           EUREKA, CA  United States      NaN       NaN          NaN   
4           Canton, OH  United States      NaN       NaN          NaN   
...                ...            ...      ...       ...          ...   
90343    Annapolis, MD  United States      NaN       NaN          NaN   
90344      Hampton, NH  United States      NaN       NaN          NaN   
90345       Payson, AZ  United States  341525N  1112021W          PAN   
90346       Morgan, UT  United States      NaN       NaN          NaN   
90347       Athens, GA  United States      NaN       NaN          NaN   

      Airport.Name  ... Purpose.of.flight         Air.carrier  \
0              NaN  ...          Personal                 NaN   
1              NaN  ...          Personal                 NaN   
2              NaN  ...          Personal                 NaN   
3              NaN  ...          Personal                 NaN   
4              NaN  ...          Personal                 NaN   
...            ...  ...               ...                 ...   
90343          NaN  ...          Personal                 NaN   
90344          NaN  ...               NaN                 NaN   
90345       PAYSON  ...          Personal                 NaN   
90346          NaN  ...          Personal  MC CESSNA 210N LLC   
90347          NaN  ...          Personal                 NaN   

      Total.Fatal.Injuries Total.Serious.Injuries Total.Minor.Injuries  \
0                      2.0                    0.0                  0.0   
1                      4.0                    0.0                  0.0   
2                      3.0                    0.0                  0.0   
3                      2.0                    0.0                  0.0   
4                      1.0                    2.0                  0.0   
...                    ...                    ...                  ...   
90343                  0.0                    1.0                  0.0   
90344                  0.0                    0.0                  0.0   
90345                  0.0                    0.0                  0.0   
90346                  0.0                    0.0                  0.0   
90347                  0.0                    1.0                  0.0   

      Total.Uninjured Weather.Condition  Broad.phase.of.flight  \
0                 0.0               UNK                 Cruise   
1                 0.0               UNK                Unknown   
2                 0.0               IMC                 Cruise   
3                 0.0               IMC                 Cruise   
4                 0.0               VMC               Approach   
...               ...               ...                    ...   
90343             0.0               NaN                    NaN   
90344             0.0               NaN                    NaN   
90345  

Next let us look at the aircraft.damage column, for the company to buy an aircraft, the type of damage needs to be low because this means less severe financial and safety consequences

In [14]:
# checking the aircraft.damage 
damage = df.loc[:, 'Aircraft.damage']
damage


0          Destroyed
1          Destroyed
2          Destroyed
3          Destroyed
4          Destroyed
            ...     
90343            NaN
90344            NaN
90345    Substantial
90346            NaN
90347            NaN
Name: Aircraft.damage, Length: 88889, dtype: object

In [15]:
# replace the nan value with unkown
df['Aircraft.damage'] = df['Aircraft.damage'].fillna('Unknown')
df

Event.Id Investigation.Type Accident.Number  Event.Date  \
0      20001218X45444           Accident      SEA87LA080  1948-10-24   
1      20001218X45447           Accident      LAX94LA336  1962-07-19   
2      20061025X01555           Accident      NYC07LA005  1974-08-30   
3      20001218X45448           Accident      LAX96LA321  1977-06-19   
4      20041105X01764           Accident      CHI79FA064  1979-08-02   
...               ...                ...             ...         ...   
90343  20221227106491           Accident      ERA23LA093  2022-12-26   
90344  20221227106494           Accident      ERA23LA095  2022-12-26   
90345  20221227106497           Accident      WPR23LA075  2022-12-26   
90346  20221227106498           Accident      WPR23LA076  2022-12-26   
90347  20221230106513           Accident      ERA23LA097  2022-12-29   

              Location        Country Latitude Longitude Airport.Code  \
0      MOOSE CREEK, ID  United States      NaN       NaN          NaN   
1       BRIDGEPORT, CA  United States      NaN       NaN          NaN   
2        Saltville, VA  United States  36.9222  -81.8781          NaN   
3           EUREKA, CA  United States      NaN       NaN          NaN   
4           Canton, OH  United States      NaN       NaN          NaN   
...                ...            ...      ...       ...          ...   
90343    Annapolis, MD  United States      NaN       NaN          NaN   
90344      Hampton, NH  United States      NaN       NaN          NaN   
90345       Payson, AZ  United States  341525N  1112021W          PAN   
90346       Morgan, UT  United States      NaN       NaN          NaN   
90347       Athens, GA  United States      NaN       NaN          NaN   

      Airport.Name  ... Purpose.of.flight         Air.carrier  \
0              NaN  ...          Personal                 NaN   
1              NaN  ...          Personal                 NaN   
2              NaN  ...          Personal                 NaN   
3              NaN  ...          Personal                 NaN   
4              NaN  ...          Personal                 NaN   
...            ...  ...               ...                 ...   
90343          NaN  ...          Personal                 NaN   
90344          NaN  ...               NaN                 NaN   
90345       PAYSON  ...          Personal                 NaN   
90346          NaN  ...          Personal  MC CESSNA 210N LLC   
90347          NaN  ...          Personal                 NaN   

      Total.Fatal.Injuries Total.Serious.Injuries Total.Minor.Injuries  \
0                      2.0                    0.0                  0.0   
1                      4.0                    0.0                  0.0   
2                      3.0                    0.0                  0.0   
3                      2.0                    0.0                  0.0   
4                      1.0                    2.0                  0.0   
...                    ...                    ...                  ...   
90343                  0.0                    1.0                  0.0   
90344                  0.0                    0.0                  0.0   
90345                  0.0                    0.0                  0.0   
90346                  0.0                    0.0                  0.0   
90347                  0.0                    1.0                  0.0   

      Total.Uninjured Weather.Condition  Broad.phase.of.flight  \
0                 0.0               UNK                 Cruise   
1                 0.0               UNK                Unknown   
2                 0.0               IMC                 Cruise   
3                 0.0               IMC                 Cruise   
4                 0.0               VMC               Approach   
...               ...               ...                    ...   
90343             0.0               NaN                    NaN   
90344             0.0               NaN                    NaN   
90345  

In [16]:
# rechecking the aircraft.damage 
damage = df.loc[:, 'Aircraft.damage']
damage

0          Destroyed
1          Destroyed
2          Destroyed
3          Destroyed
4          Destroyed
            ...     
90343        Unknown
90344        Unknown
90345    Substantial
90346        Unknown
90347        Unknown
Name: Aircraft.damage, Length: 88889, dtype: object

Now that we have seen the type of damages that the aircraft encounter, next we look at the categories of aircraft we have because from our problem statement, the company want to purchase and operate airplanes for <b>commercial and private enterprises</b> meaning our focus will be inclined to that direction

In [18]:
# checking the aircraft category we are having 
category = df.loc[:, 'Aircraft.Category'].unique()
category

array([nan, 'Airplane', 'Helicopter', 'Glider', 'Balloon', 'Gyrocraft',
       'Ultralight', 'Unknown', 'Blimp', 'Powered-Lift', 'Weight-Shift',
       'Powered Parachute', 'Rocket', 'WSFT', 'UNK', 'ULTR'], dtype=object)

From the above info we get to see the types of aircraft that is available. Also we can see there are some null values in our dataset, so instead of dropping them, let's just replace the null values with unknown cause already there are some aircraft that their category is unknown 

In [ ]:
# replacing the null values with unknown
df['Aircraft.Category'] = df['Aircraft.Category'].fillna('Unknown')
df

Event.Id Investigation.Type Accident.Number  Event.Date  \
0      20001218X45444           Accident      SEA87LA080  1948-10-24   
1      20001218X45447           Accident      LAX94LA336  1962-07-19   
2      20061025X01555           Accident      NYC07LA005  1974-08-30   
3      20001218X45448           Accident      LAX96LA321  1977-06-19   
4      20041105X01764           Accident      CHI79FA064  1979-08-02   
...               ...                ...             ...         ...   
90343  20221227106491           Accident      ERA23LA093  2022-12-26   
90344  20221227106494           Accident      ERA23LA095  2022-12-26   
90345  20221227106497           Accident      WPR23LA075  2022-12-26   
90346  20221227106498           Accident      WPR23LA076  2022-12-26   
90347  20221230106513           Accident      ERA23LA097  2022-12-29   

              Location        Country Latitude Longitude Airport.Code  \
0      MOOSE CREEK, ID  United States      NaN       NaN          NaN   
1       BRIDGEPORT, CA  United States      NaN       NaN          NaN   
2        Saltville, VA  United States  36.9222  -81.8781          NaN   
3           EUREKA, CA  United States      NaN       NaN          NaN   
4           Canton, OH  United States      NaN       NaN          NaN   
...                ...            ...      ...       ...          ...   
90343    Annapolis, MD  United States      NaN       NaN          NaN   
90344      Hampton, NH  United States      NaN       NaN          NaN   
90345       Payson, AZ  United States  341525N  1112021W          PAN   
90346       Morgan, UT  United States      NaN       NaN          NaN   
90347       Athens, GA  United States      NaN       NaN          NaN   

      Airport.Name  ... Purpose.of.flight         Air.carrier  \
0              NaN  ...          Personal                 NaN   
1              NaN  ...          Personal                 NaN   
2              NaN  ...          Personal                 NaN   
3              NaN  ...          Personal                 NaN   
4              NaN  ...          Personal                 NaN   
...            ...  ...               ...                 ...   
90343          NaN  ...          Personal                 NaN   
90344          NaN  ...               NaN                 NaN   
90345       PAYSON  ...          Personal                 NaN   
90346          NaN  ...          Personal  MC CESSNA 210N LLC   
90347          NaN  ...          Personal                 NaN   

      Total.Fatal.Injuries Total.Serious.Injuries Total.Minor.Injuries  \
0                      2.0                    0.0                  0.0   
1                      4.0                    0.0                  0.0   
2                      3.0                    0.0                  0.0   
3                      2.0                    0.0                  0.0   
4                      1.0                    2.0                  0.0   
...                    ...                    ...                  ...   
90343                  0.0                    1.0                  0.0   
90344                  0.0                    0.0                  0.0   
90345                  0.0                    0.0                  0.0   
90346                  0.0                    0.0                  0.0   
90347                  0.0                    1.0                  0.0   

      Total.Uninjured Weather.Condition  Broad.phase.of.flight  \
0                 0.0               UNK                 Cruise   
1                 0.0               UNK                Unknown   
2                 0.0               IMC                 Cruise   
3                 0.0               IMC                 Cruise   
4                 0.0               VMC               Approach   
...               ...               ...                    ...   
90343             0.0               NaN                    NaN   
90344             0.0               NaN                    NaN   
90345  

In [20]:
# rechecking the aircraft category we are having 
category = df.loc[:, 'Aircraft.Category'].unique()
category

array(['Unknown', 'Airplane', 'Helicopter', 'Glider', 'Balloon',
       'Gyrocraft', 'Ultralight', 'Blimp', 'Powered-Lift', 'Weight-Shift',
       'Powered Parachute', 'Rocket', 'WSFT', 'UNK', 'ULTR'], dtype=object)

Next we look at the type of built the availale aircraft fall into. Most companies will opt to go for the built that aint an amateur especially for commercial use. Safety needs to be guaranteed

In [23]:
# checking for the built of the available aircrafts
built = df.loc[:, 'Amateur.Built'].unique()
built

array(['No', 'Yes', nan], dtype=object)

In [ ]:
# replacing the nan values with unknown
df['Amateur.Built'] = df['Amateur.Built'].fillna('Unknown')
df

Event.Id Investigation.Type Accident.Number  Event.Date  \
0      20001218X45444           Accident      SEA87LA080  1948-10-24   
1      20001218X45447           Accident      LAX94LA336  1962-07-19   
2      20061025X01555           Accident      NYC07LA005  1974-08-30   
3      20001218X45448           Accident      LAX96LA321  1977-06-19   
4      20041105X01764           Accident      CHI79FA064  1979-08-02   
...               ...                ...             ...         ...   
90343  20221227106491           Accident      ERA23LA093  2022-12-26   
90344  20221227106494           Accident      ERA23LA095  2022-12-26   
90345  20221227106497           Accident      WPR23LA075  2022-12-26   
90346  20221227106498           Accident      WPR23LA076  2022-12-26   
90347  20221230106513           Accident      ERA23LA097  2022-12-29   

              Location        Country Latitude Longitude Airport.Code  \
0      MOOSE CREEK, ID  United States      NaN       NaN          NaN   
1       BRIDGEPORT, CA  United States      NaN       NaN          NaN   
2        Saltville, VA  United States  36.9222  -81.8781          NaN   
3           EUREKA, CA  United States      NaN       NaN          NaN   
4           Canton, OH  United States      NaN       NaN          NaN   
...                ...            ...      ...       ...          ...   
90343    Annapolis, MD  United States      NaN       NaN          NaN   
90344      Hampton, NH  United States      NaN       NaN          NaN   
90345       Payson, AZ  United States  341525N  1112021W          PAN   
90346       Morgan, UT  United States      NaN       NaN          NaN   
90347       Athens, GA  United States      NaN       NaN          NaN   

      Airport.Name  ... Purpose.of.flight         Air.carrier  \
0              NaN  ...          Personal                 NaN   
1              NaN  ...          Personal                 NaN   
2              NaN  ...          Personal                 NaN   
3              NaN  ...          Personal                 NaN   
4              NaN  ...          Personal                 NaN   
...            ...  ...               ...                 ...   
90343          NaN  ...          Personal                 NaN   
90344          NaN  ...               NaN                 NaN   
90345       PAYSON  ...          Personal                 NaN   
90346          NaN  ...          Personal  MC CESSNA 210N LLC   
90347          NaN  ...          Personal                 NaN   

      Total.Fatal.Injuries Total.Serious.Injuries Total.Minor.Injuries  \
0                      2.0                    0.0                  0.0   
1                      4.0                    0.0                  0.0   
2                      3.0                    0.0                  0.0   
3                      2.0                    0.0                  0.0   
4                      1.0                    2.0                  0.0   
...                    ...                    ...                  ...   
90343                  0.0                    1.0                  0.0   
90344                  0.0                    0.0                  0.0   
90345                  0.0                    0.0                  0.0   
90346                  0.0                    0.0                  0.0   
90347                  0.0                    1.0                  0.0   

      Total.Uninjured Weather.Condition  Broad.phase.of.flight  \
0                 0.0               UNK                 Cruise   
1                 0.0               UNK                Unknown   
2                 0.0               IMC                 Cruise   
3                 0.0               IMC                 Cruise   
4                 0.0               VMC               Approach   
...               ...               ...                    ...   
90343             0.0               NaN                    NaN   
90344             0.0               NaN                    NaN   
90345  

Next we are going to look a the Purpose.of.flight. Sinced the company wants to bu aircraft for commercial and private purpose,we need to clean the data in this column remove the nun values so that we use the data later

In [29]:
# checking for the purpose of the current aircrafts
purpose = df.loc[:, 'Purpose.of.flight'].unique()
purpose

array(['Personal', nan, 'Business', 'Instructional', 'Unknown', 'Ferry',
       'Executive/corporate', 'Aerial Observation', 'Aerial Application',
       'Public Aircraft', 'Skydiving', 'Other Work Use', 'Positioning',
       'Flight Test', 'Air Race/show', 'Air Drop',
       'Public Aircraft - Federal', 'Glider Tow',
       'Public Aircraft - Local', 'External Load',
       'Public Aircraft - State', 'Banner Tow', 'Firefighting',
       'Air Race show', 'PUBS', 'ASHO', 'PUBL'], dtype=object)

In [30]:
# replacing the nan values with unknown
df['Purpose.of.flight'] = df['Purpose.of.flight'].fillna('Unknown')
df

Event.Id Investigation.Type Accident.Number  Event.Date  \
0      20001218X45444           Accident      SEA87LA080  1948-10-24   
1      20001218X45447           Accident      LAX94LA336  1962-07-19   
2      20061025X01555           Accident      NYC07LA005  1974-08-30   
3      20001218X45448           Accident      LAX96LA321  1977-06-19   
4      20041105X01764           Accident      CHI79FA064  1979-08-02   
...               ...                ...             ...         ...   
90343  20221227106491           Accident      ERA23LA093  2022-12-26   
90344  20221227106494           Accident      ERA23LA095  2022-12-26   
90345  20221227106497           Accident      WPR23LA075  2022-12-26   
90346  20221227106498           Accident      WPR23LA076  2022-12-26   
90347  20221230106513           Accident      ERA23LA097  2022-12-29   

              Location        Country Latitude Longitude Airport.Code  \
0      MOOSE CREEK, ID  United States      NaN       NaN          NaN   
1       BRIDGEPORT, CA  United States      NaN       NaN          NaN   
2        Saltville, VA  United States  36.9222  -81.8781          NaN   
3           EUREKA, CA  United States      NaN       NaN          NaN   
4           Canton, OH  United States      NaN       NaN          NaN   
...                ...            ...      ...       ...          ...   
90343    Annapolis, MD  United States      NaN       NaN          NaN   
90344      Hampton, NH  United States      NaN       NaN          NaN   
90345       Payson, AZ  United States  341525N  1112021W          PAN   
90346       Morgan, UT  United States      NaN       NaN          NaN   
90347       Athens, GA  United States      NaN       NaN          NaN   

      Airport.Name  ... Purpose.of.flight         Air.carrier  \
0              NaN  ...          Personal                 NaN   
1              NaN  ...          Personal                 NaN   
2              NaN  ...          Personal                 NaN   
3              NaN  ...          Personal                 NaN   
4              NaN  ...          Personal                 NaN   
...            ...  ...               ...                 ...   
90343          NaN  ...          Personal                 NaN   
90344          NaN  ...           Unknown                 NaN   
90345       PAYSON  ...          Personal                 NaN   
90346          NaN  ...          Personal  MC CESSNA 210N LLC   
90347          NaN  ...          Personal                 NaN   

      Total.Fatal.Injuries Total.Serious.Injuries Total.Minor.Injuries  \
0                      2.0                    0.0                  0.0   
1                      4.0                    0.0                  0.0   
2                      3.0                    0.0                  0.0   
3                      2.0                    0.0                  0.0   
4                      1.0                    2.0                  0.0   
...                    ...                    ...                  ...   
90343                  0.0                    1.0                  0.0   
90344                  0.0                    0.0                  0.0   
90345                  0.0                    0.0                  0.0   
90346                  0.0                    0.0                  0.0   
90347                  0.0                    1.0                  0.0   

      Total.Uninjured Weather.Condition  Broad.phase.of.flight  \
0                 0.0               UNK                 Cruise   
1                 0.0               UNK                Unknown   
2                 0.0               IMC                 Cruise   
3                 0.0               IMC                 Cruise   
4                 0.0               VMC               Approach   
...               ...               ...                    ...   
90343             0.0               NaN                    NaN   
90344             0.0               NaN                    NaN   
90345  

In [31]:
# rechecking for the purpose of the current aircrafts
purpose = df.loc[:, 'Purpose.of.flight'].unique()
purpose

array(['Personal', 'Unknown', 'Business', 'Instructional', 'Ferry',
       'Executive/corporate', 'Aerial Observation', 'Aerial Application',
       'Public Aircraft', 'Skydiving', 'Other Work Use', 'Positioning',
       'Flight Test', 'Air Race/show', 'Air Drop',
       'Public Aircraft - Federal', 'Glider Tow',
       'Public Aircraft - Local', 'External Load',
       'Public Aircraft - State', 'Banner Tow', 'Firefighting',
       'Air Race show', 'PUBS', 'ASHO', 'PUBL'], dtype=object)

The next column we need to consider is also the no.of.engine column. Buyers will always want to purchase an aircraft with multiple engine because multi engine aircraft are often perceived as lower risk for certain missions due to redundancy

In [32]:
# checking for the engine of the current aircrafts
no_engine = df.loc[:, 'Number.of.Engines'].unique()
no_engine

array([ 1., nan,  2.,  0.,  3.,  4.,  8.,  6.])

In [33]:
# replacing the nan values with 0
df['Number.of.Engines'] = df['Number.of.Engines'].fillna(0)
df

Event.Id Investigation.Type Accident.Number  Event.Date  \
0      20001218X45444           Accident      SEA87LA080  1948-10-24   
1      20001218X45447           Accident      LAX94LA336  1962-07-19   
2      20061025X01555           Accident      NYC07LA005  1974-08-30   
3      20001218X45448           Accident      LAX96LA321  1977-06-19   
4      20041105X01764           Accident      CHI79FA064  1979-08-02   
...               ...                ...             ...         ...   
90343  20221227106491           Accident      ERA23LA093  2022-12-26   
90344  20221227106494           Accident      ERA23LA095  2022-12-26   
90345  20221227106497           Accident      WPR23LA075  2022-12-26   
90346  20221227106498           Accident      WPR23LA076  2022-12-26   
90347  20221230106513           Accident      ERA23LA097  2022-12-29   

              Location        Country Latitude Longitude Airport.Code  \
0      MOOSE CREEK, ID  United States      NaN       NaN          NaN   
1       BRIDGEPORT, CA  United States      NaN       NaN          NaN   
2        Saltville, VA  United States  36.9222  -81.8781          NaN   
3           EUREKA, CA  United States      NaN       NaN          NaN   
4           Canton, OH  United States      NaN       NaN          NaN   
...                ...            ...      ...       ...          ...   
90343    Annapolis, MD  United States      NaN       NaN          NaN   
90344      Hampton, NH  United States      NaN       NaN          NaN   
90345       Payson, AZ  United States  341525N  1112021W          PAN   
90346       Morgan, UT  United States      NaN       NaN          NaN   
90347       Athens, GA  United States      NaN       NaN          NaN   

      Airport.Name  ... Purpose.of.flight         Air.carrier  \
0              NaN  ...          Personal                 NaN   
1              NaN  ...          Personal                 NaN   
2              NaN  ...          Personal                 NaN   
3              NaN  ...          Personal                 NaN   
4              NaN  ...          Personal                 NaN   
...            ...  ...               ...                 ...   
90343          NaN  ...          Personal                 NaN   
90344          NaN  ...           Unknown                 NaN   
90345       PAYSON  ...          Personal                 NaN   
90346          NaN  ...          Personal  MC CESSNA 210N LLC   
90347          NaN  ...          Personal                 NaN   

      Total.Fatal.Injuries Total.Serious.Injuries Total.Minor.Injuries  \
0                      2.0                    0.0                  0.0   
1                      4.0                    0.0                  0.0   
2                      3.0                    0.0                  0.0   
3                      2.0                    0.0                  0.0   
4                      1.0                    2.0                  0.0   
...                    ...                    ...                  ...   
90343                  0.0                    1.0                  0.0   
90344                  0.0                    0.0                  0.0   
90345                  0.0                    0.0                  0.0   
90346                  0.0                    0.0                  0.0   
90347                  0.0                    1.0                  0.0   

      Total.Uninjured Weather.Condition  Broad.phase.of.flight  \
0                 0.0               UNK                 Cruise   
1                 0.0               UNK                Unknown   
2                 0.0               IMC                 Cruise   
3                 0.0               IMC                 Cruise   
4                 0.0               VMC               Approach   
...               ...               ...                    ...   
90343             0.0               NaN                    NaN   
90344             0.0               NaN                    NaN   
90345  

In [34]:
# rechecking for the engine of the aircrafts
no_engine = df.loc[:, 'Number.of.Engines'].unique()
no_engine

array([1., 0., 2., 3., 4., 8., 6.])

Another thing that the buyers wil consider is the make and model of aircraft. Buyers will be looking for a certain type of model or make since they do believe they operate perfectly, so wewill look thta its data replace the nan values if present and store our new details to be used later on

In [ ]:
# checking the current model for the aircrafts
model = df.loc[:, 'Model'].unique()
model

array(['108-3', 'PA24-180', '172M', ..., 'ROTORWAY EXEC 162-F',
       'KITFOX S5', 'M-8 EAGLE'], dtype=object)

As much as it cannot be seen above that ther are missing values in this column, at the very top among the first 3 codes we run, we got to see there were 1551 missing data in this column so just like the rest we are going to clean that data up and replace the empty one with unknown.

In [36]:
# replacing the nan values for the model types of the aircrafts
df['Model'] = df['Model'].fillna('Unknown')
df


Event.Id Investigation.Type Accident.Number  Event.Date  \
0      20001218X45444           Accident      SEA87LA080  1948-10-24   
1      20001218X45447           Accident      LAX94LA336  1962-07-19   
2      20061025X01555           Accident      NYC07LA005  1974-08-30   
3      20001218X45448           Accident      LAX96LA321  1977-06-19   
4      20041105X01764           Accident      CHI79FA064  1979-08-02   
...               ...                ...             ...         ...   
90343  20221227106491           Accident      ERA23LA093  2022-12-26   
90344  20221227106494           Accident      ERA23LA095  2022-12-26   
90345  20221227106497           Accident      WPR23LA075  2022-12-26   
90346  20221227106498           Accident      WPR23LA076  2022-12-26   
90347  20221230106513           Accident      ERA23LA097  2022-12-29   

              Location        Country Latitude Longitude Airport.Code  \
0      MOOSE CREEK, ID  United States      NaN       NaN          NaN   
1       BRIDGEPORT, CA  United States      NaN       NaN          NaN   
2        Saltville, VA  United States  36.9222  -81.8781          NaN   
3           EUREKA, CA  United States      NaN       NaN          NaN   
4           Canton, OH  United States      NaN       NaN          NaN   
...                ...            ...      ...       ...          ...   
90343    Annapolis, MD  United States      NaN       NaN          NaN   
90344      Hampton, NH  United States      NaN       NaN          NaN   
90345       Payson, AZ  United States  341525N  1112021W          PAN   
90346       Morgan, UT  United States      NaN       NaN          NaN   
90347       Athens, GA  United States      NaN       NaN          NaN   

      Airport.Name  ... Purpose.of.flight         Air.carrier  \
0              NaN  ...          Personal                 NaN   
1              NaN  ...          Personal                 NaN   
2              NaN  ...          Personal                 NaN   
3              NaN  ...          Personal                 NaN   
4              NaN  ...          Personal                 NaN   
...            ...  ...               ...                 ...   
90343          NaN  ...          Personal                 NaN   
90344          NaN  ...           Unknown                 NaN   
90345       PAYSON  ...          Personal                 NaN   
90346          NaN  ...          Personal  MC CESSNA 210N LLC   
90347          NaN  ...          Personal                 NaN   

      Total.Fatal.Injuries Total.Serious.Injuries Total.Minor.Injuries  \
0                      2.0                    0.0                  0.0   
1                      4.0                    0.0                  0.0   
2                      3.0                    0.0                  0.0   
3                      2.0                    0.0                  0.0   
4                      1.0                    2.0                  0.0   
...                    ...                    ...                  ...   
90343                  0.0                    1.0                  0.0   
90344                  0.0                    0.0                  0.0   
90345                  0.0                    0.0                  0.0   
90346                  0.0                    0.0                  0.0   
90347                  0.0                    1.0                  0.0   

      Total.Uninjured Weather.Condition  Broad.phase.of.flight  \
0                 0.0               UNK                 Cruise   
1                 0.0               UNK                Unknown   
2                 0.0               IMC                 Cruise   
3                 0.0               IMC                 Cruise   
4                 0.0               VMC               Approach   
...               ...               ...                    ...   
90343             0.0               NaN                    NaN   
90344             0.0               NaN                    NaN   
90345  

In [37]:
# rechecking with the current model for the aircrafts
model = df.loc[:, 'Model'].unique()
model

array(['108-3', 'PA24-180', '172M', ..., 'ROTORWAY EXEC 162-F',
       'KITFOX S5', 'M-8 EAGLE'], dtype=object)

We do the same for make and engine type

In [38]:
# checking the makes of the aircrafts
make = df.loc[:, 'Make'].unique()
make

array(['Stinson', 'Piper', 'Cessna', ..., 'JAMES R DERNOVSEK',
       'ORLICAN S R O', 'ROYSE RALPH L'], dtype=object)

In [39]:
# replacing the nan values for the makes of the aircrafts
df['Make'] = df['Make'].fillna('Unknown')
df

Event.Id Investigation.Type Accident.Number  Event.Date  \
0      20001218X45444           Accident      SEA87LA080  1948-10-24   
1      20001218X45447           Accident      LAX94LA336  1962-07-19   
2      20061025X01555           Accident      NYC07LA005  1974-08-30   
3      20001218X45448           Accident      LAX96LA321  1977-06-19   
4      20041105X01764           Accident      CHI79FA064  1979-08-02   
...               ...                ...             ...         ...   
90343  20221227106491           Accident      ERA23LA093  2022-12-26   
90344  20221227106494           Accident      ERA23LA095  2022-12-26   
90345  20221227106497           Accident      WPR23LA075  2022-12-26   
90346  20221227106498           Accident      WPR23LA076  2022-12-26   
90347  20221230106513           Accident      ERA23LA097  2022-12-29   

              Location        Country Latitude Longitude Airport.Code  \
0      MOOSE CREEK, ID  United States      NaN       NaN          NaN   
1       BRIDGEPORT, CA  United States      NaN       NaN          NaN   
2        Saltville, VA  United States  36.9222  -81.8781          NaN   
3           EUREKA, CA  United States      NaN       NaN          NaN   
4           Canton, OH  United States      NaN       NaN          NaN   
...                ...            ...      ...       ...          ...   
90343    Annapolis, MD  United States      NaN       NaN          NaN   
90344      Hampton, NH  United States      NaN       NaN          NaN   
90345       Payson, AZ  United States  341525N  1112021W          PAN   
90346       Morgan, UT  United States      NaN       NaN          NaN   
90347       Athens, GA  United States      NaN       NaN          NaN   

      Airport.Name  ... Purpose.of.flight         Air.carrier  \
0              NaN  ...          Personal                 NaN   
1              NaN  ...          Personal                 NaN   
2              NaN  ...          Personal                 NaN   
3              NaN  ...          Personal                 NaN   
4              NaN  ...          Personal                 NaN   
...            ...  ...               ...                 ...   
90343          NaN  ...          Personal                 NaN   
90344          NaN  ...           Unknown                 NaN   
90345       PAYSON  ...          Personal                 NaN   
90346          NaN  ...          Personal  MC CESSNA 210N LLC   
90347          NaN  ...          Personal                 NaN   

      Total.Fatal.Injuries Total.Serious.Injuries Total.Minor.Injuries  \
0                      2.0                    0.0                  0.0   
1                      4.0                    0.0                  0.0   
2                      3.0                    0.0                  0.0   
3                      2.0                    0.0                  0.0   
4                      1.0                    2.0                  0.0   
...                    ...                    ...                  ...   
90343                  0.0                    1.0                  0.0   
90344                  0.0                    0.0                  0.0   
90345                  0.0                    0.0                  0.0   
90346                  0.0                    0.0                  0.0   
90347                  0.0                    1.0                  0.0   

      Total.Uninjured Weather.Condition  Broad.phase.of.flight  \
0                 0.0               UNK                 Cruise   
1                 0.0               UNK                Unknown   
2                 0.0               IMC                 Cruise   
3                 0.0               IMC                 Cruise   
4                 0.0               VMC               Approach   
...               ...               ...                    ...   
90343             0.0               NaN                    NaN   
90344             0.0               NaN                    NaN   
90345  

In [40]:
# rechecking the makes of the aircrafts
make = df.loc[:, 'Make'].unique()
make

array(['Stinson', 'Piper', 'Cessna', ..., 'JAMES R DERNOVSEK',
       'ORLICAN S R O', 'ROYSE RALPH L'], dtype=object)

In [41]:
# checking the engine type of the aircrafts
engine_type= df.loc[:, 'Engine.Type'].unique()
engine_type

array(['Reciprocating', nan, 'Turbo Fan', 'Turbo Shaft', 'Unknown',
       'Turbo Prop', 'Turbo Jet', 'None', 'Electric', 'Hybrid Rocket',
       'Geared Turbofan', 'LR', 'NONE', 'UNK'], dtype=object)

In [42]:
# replacing the nan values for the engine types of the aircrafts
df['Engine.Type'] = df['Engine.Type'].fillna('Unknown')
df

Event.Id Investigation.Type Accident.Number  Event.Date  \
0      20001218X45444           Accident      SEA87LA080  1948-10-24   
1      20001218X45447           Accident      LAX94LA336  1962-07-19   
2      20061025X01555           Accident      NYC07LA005  1974-08-30   
3      20001218X45448           Accident      LAX96LA321  1977-06-19   
4      20041105X01764           Accident      CHI79FA064  1979-08-02   
...               ...                ...             ...         ...   
90343  20221227106491           Accident      ERA23LA093  2022-12-26   
90344  20221227106494           Accident      ERA23LA095  2022-12-26   
90345  20221227106497           Accident      WPR23LA075  2022-12-26   
90346  20221227106498           Accident      WPR23LA076  2022-12-26   
90347  20221230106513           Accident      ERA23LA097  2022-12-29   

              Location        Country Latitude Longitude Airport.Code  \
0      MOOSE CREEK, ID  United States      NaN       NaN          NaN   
1       BRIDGEPORT, CA  United States      NaN       NaN          NaN   
2        Saltville, VA  United States  36.9222  -81.8781          NaN   
3           EUREKA, CA  United States      NaN       NaN          NaN   
4           Canton, OH  United States      NaN       NaN          NaN   
...                ...            ...      ...       ...          ...   
90343    Annapolis, MD  United States      NaN       NaN          NaN   
90344      Hampton, NH  United States      NaN       NaN          NaN   
90345       Payson, AZ  United States  341525N  1112021W          PAN   
90346       Morgan, UT  United States      NaN       NaN          NaN   
90347       Athens, GA  United States      NaN       NaN          NaN   

      Airport.Name  ... Purpose.of.flight         Air.carrier  \
0              NaN  ...          Personal                 NaN   
1              NaN  ...          Personal                 NaN   
2              NaN  ...          Personal                 NaN   
3              NaN  ...          Personal                 NaN   
4              NaN  ...          Personal                 NaN   
...            ...  ...               ...                 ...   
90343          NaN  ...          Personal                 NaN   
90344          NaN  ...           Unknown                 NaN   
90345       PAYSON  ...          Personal                 NaN   
90346          NaN  ...          Personal  MC CESSNA 210N LLC   
90347          NaN  ...          Personal                 NaN   

      Total.Fatal.Injuries Total.Serious.Injuries Total.Minor.Injuries  \
0                      2.0                    0.0                  0.0   
1                      4.0                    0.0                  0.0   
2                      3.0                    0.0                  0.0   
3                      2.0                    0.0                  0.0   
4                      1.0                    2.0                  0.0   
...                    ...                    ...                  ...   
90343                  0.0                    1.0                  0.0   
90344                  0.0                    0.0                  0.0   
90345                  0.0                    0.0                  0.0   
90346                  0.0                    0.0                  0.0   
90347                  0.0                    1.0                  0.0   

      Total.Uninjured Weather.Condition  Broad.phase.of.flight  \
0                 0.0               UNK                 Cruise   
1                 0.0               UNK                Unknown   
2                 0.0               IMC                 Cruise   
3                 0.0               IMC                 Cruise   
4                 0.0               VMC               Approach   
...               ...               ...                    ...   
90343             0.0               NaN                    NaN   
90344             0.0               NaN                    NaN   
90345  

In [43]:
# rechecking the engine type of the aircrafts
engine_type= df.loc[:, 'Engine.Type'].unique()
engine_type

array(['Reciprocating', 'Unknown', 'Turbo Fan', 'Turbo Shaft',
       'Turbo Prop', 'Turbo Jet', 'None', 'Electric', 'Hybrid Rocket',
       'Geared Turbofan', 'LR', 'NONE', 'UNK'], dtype=object)

Another thig that buyers will consider is also the weather condition. The type of weather condition at a certain place will determine whether an aircraft is goint to with stand the weather or an accident occur. This condition also work hand in hand with the make and model o the aircraft. So we will do what we had done above clean the data and replace the empty columns with the appropriate values

In [44]:
# checking on the type of weather condition an aircraft is subjeted to
weather_cond = df.loc[:, 'Weather.Condition'].unique()
weather_cond

array(['UNK', 'IMC', 'VMC', nan, 'Unk'], dtype=object)

In [45]:
# replacet the nan with unknown
df['Weather.Condition'] = df['Weather.Condition'].fillna('Unknown')
df

Event.Id Investigation.Type Accident.Number  Event.Date  \
0      20001218X45444           Accident      SEA87LA080  1948-10-24   
1      20001218X45447           Accident      LAX94LA336  1962-07-19   
2      20061025X01555           Accident      NYC07LA005  1974-08-30   
3      20001218X45448           Accident      LAX96LA321  1977-06-19   
4      20041105X01764           Accident      CHI79FA064  1979-08-02   
...               ...                ...             ...         ...   
90343  20221227106491           Accident      ERA23LA093  2022-12-26   
90344  20221227106494           Accident      ERA23LA095  2022-12-26   
90345  20221227106497           Accident      WPR23LA075  2022-12-26   
90346  20221227106498           Accident      WPR23LA076  2022-12-26   
90347  20221230106513           Accident      ERA23LA097  2022-12-29   

              Location        Country Latitude Longitude Airport.Code  \
0      MOOSE CREEK, ID  United States      NaN       NaN          NaN   
1       BRIDGEPORT, CA  United States      NaN       NaN          NaN   
2        Saltville, VA  United States  36.9222  -81.8781          NaN   
3           EUREKA, CA  United States      NaN       NaN          NaN   
4           Canton, OH  United States      NaN       NaN          NaN   
...                ...            ...      ...       ...          ...   
90343    Annapolis, MD  United States      NaN       NaN          NaN   
90344      Hampton, NH  United States      NaN       NaN          NaN   
90345       Payson, AZ  United States  341525N  1112021W          PAN   
90346       Morgan, UT  United States      NaN       NaN          NaN   
90347       Athens, GA  United States      NaN       NaN          NaN   

      Airport.Name  ... Purpose.of.flight         Air.carrier  \
0              NaN  ...          Personal                 NaN   
1              NaN  ...          Personal                 NaN   
2              NaN  ...          Personal                 NaN   
3              NaN  ...          Personal                 NaN   
4              NaN  ...          Personal                 NaN   
...            ...  ...               ...                 ...   
90343          NaN  ...          Personal                 NaN   
90344          NaN  ...           Unknown                 NaN   
90345       PAYSON  ...          Personal                 NaN   
90346          NaN  ...          Personal  MC CESSNA 210N LLC   
90347          NaN  ...          Personal                 NaN   

      Total.Fatal.Injuries Total.Serious.Injuries Total.Minor.Injuries  \
0                      2.0                    0.0                  0.0   
1                      4.0                    0.0                  0.0   
2                      3.0                    0.0                  0.0   
3                      2.0                    0.0                  0.0   
4                      1.0                    2.0                  0.0   
...                    ...                    ...                  ...   
90343                  0.0                    1.0                  0.0   
90344                  0.0                    0.0                  0.0   
90345                  0.0                    0.0                  0.0   
90346                  0.0                    0.0                  0.0   
90347                  0.0                    1.0                  0.0   

      Total.Uninjured Weather.Condition  Broad.phase.of.flight  \
0                 0.0               UNK                 Cruise   
1                 0.0               UNK                Unknown   
2                 0.0               IMC                 Cruise   
3                 0.0               IMC                 Cruise   
4                 0.0               VMC               Approach   
...               ...               ...                    ...   
90343             0.0           Unknown                    NaN   
90344             0.0           Unknown                    NaN   
90345  

In [46]:
# rechecking the type of weather condition an aircraft is subjeted to
weather_cond = df.loc[:, 'Weather.Condition'].unique()
weather_cond

array(['UNK', 'IMC', 'VMC', 'Unknown', 'Unk'], dtype=object)

Lastly on the cleaning of the data of the columns that will help us solve outr problem statement is Broad.phase.of.flight. This will help us identify a model that ha a specific weakness

In [47]:
# checking the Broad.phase.of flight
phase = df.loc[:, 'Broad.phase.of.flight'].unique()
phase

array(['Cruise', 'Unknown', 'Approach', 'Climb', 'Takeoff', 'Landing',
       'Taxi', 'Descent', 'Maneuvering', 'Standing', 'Go-around', 'Other',
       nan], dtype=object)

In [48]:
# replacing the nan with unknown
df['Broad.phase.of.flight'] = df['Broad.phase.of.flight'].fillna('Unknown')
df

Event.Id Investigation.Type Accident.Number  Event.Date  \
0      20001218X45444           Accident      SEA87LA080  1948-10-24   
1      20001218X45447           Accident      LAX94LA336  1962-07-19   
2      20061025X01555           Accident      NYC07LA005  1974-08-30   
3      20001218X45448           Accident      LAX96LA321  1977-06-19   
4      20041105X01764           Accident      CHI79FA064  1979-08-02   
...               ...                ...             ...         ...   
90343  20221227106491           Accident      ERA23LA093  2022-12-26   
90344  20221227106494           Accident      ERA23LA095  2022-12-26   
90345  20221227106497           Accident      WPR23LA075  2022-12-26   
90346  20221227106498           Accident      WPR23LA076  2022-12-26   
90347  20221230106513           Accident      ERA23LA097  2022-12-29   

              Location        Country Latitude Longitude Airport.Code  \
0      MOOSE CREEK, ID  United States      NaN       NaN          NaN   
1       BRIDGEPORT, CA  United States      NaN       NaN          NaN   
2        Saltville, VA  United States  36.9222  -81.8781          NaN   
3           EUREKA, CA  United States      NaN       NaN          NaN   
4           Canton, OH  United States      NaN       NaN          NaN   
...                ...            ...      ...       ...          ...   
90343    Annapolis, MD  United States      NaN       NaN          NaN   
90344      Hampton, NH  United States      NaN       NaN          NaN   
90345       Payson, AZ  United States  341525N  1112021W          PAN   
90346       Morgan, UT  United States      NaN       NaN          NaN   
90347       Athens, GA  United States      NaN       NaN          NaN   

      Airport.Name  ... Purpose.of.flight         Air.carrier  \
0              NaN  ...          Personal                 NaN   
1              NaN  ...          Personal                 NaN   
2              NaN  ...          Personal                 NaN   
3              NaN  ...          Personal                 NaN   
4              NaN  ...          Personal                 NaN   
...            ...  ...               ...                 ...   
90343          NaN  ...          Personal                 NaN   
90344          NaN  ...           Unknown                 NaN   
90345       PAYSON  ...          Personal                 NaN   
90346          NaN  ...          Personal  MC CESSNA 210N LLC   
90347          NaN  ...          Personal                 NaN   

      Total.Fatal.Injuries Total.Serious.Injuries Total.Minor.Injuries  \
0                      2.0                    0.0                  0.0   
1                      4.0                    0.0                  0.0   
2                      3.0                    0.0                  0.0   
3                      2.0                    0.0                  0.0   
4                      1.0                    2.0                  0.0   
...                    ...                    ...                  ...   
90343                  0.0                    1.0                  0.0   
90344                  0.0                    0.0                  0.0   
90345                  0.0                    0.0                  0.0   
90346                  0.0                    0.0                  0.0   
90347                  0.0                    1.0                  0.0   

      Total.Uninjured Weather.Condition  Broad.phase.of.flight  \
0                 0.0               UNK                 Cruise   
1                 0.0               UNK                Unknown   
2                 0.0               IMC                 Cruise   
3                 0.0               IMC                 Cruise   
4                 0.0               VMC               Approach   
...               ...               ...                    ...   
90343             0.0           Unknown                Unknown   
90344             0.0           Unknown                Unknown   
90345  

In [49]:
# rechecking the Broad.phase.of flight
phase = df.loc[:, 'Broad.phase.of.flight'].unique()
phase

array(['Cruise', 'Unknown', 'Approach', 'Climb', 'Takeoff', 'Landing',
       'Taxi', 'Descent', 'Maneuvering', 'Standing', 'Go-around', 'Other'],
      dtype=object)

In [50]:
df.isna().sum()

Event.Id                      0
Investigation.Type            0
Accident.Number               0
Event.Date                    0
Location                     52
Country                     226
Latitude                  54507
Longitude                 54516
Airport.Code              38640
Airport.Name              36099
Injury.Severity               0
Aircraft.damage               0
Aircraft.Category             0
Registration.Number        1317
Make                          0
Model                         0
Amateur.Built                 0
Number.of.Engines             0
Engine.Type                   0
FAR.Description           56866
Schedule                  76307
Purpose.of.flight             0
Air.carrier               72241
Total.Fatal.Injuries          0
Total.Serious.Injuries        0
Total.Minor.Injuries          0
Total.Uninjured               0
Weather.Condition             0
Broad.phase.of.flight         0
Report.Status              6381
Publication.Date          15230
dtype: i

When it comes to the model of the aircraft as much as the missing values had been removed, there are cases of both uppercase and lower case being used to refere to one specific model, so the next step is to remove such occurence

In [ ]:
# Make and Model names
df['Make'] = df['Make'].str.upper().str.strip()
df['Model'] = df['Model'].str.upper().str.strip()